python/metatensor-learn/metatensor/learn/nn/module_map.py

```diff
@@ -280,7 +282,13 @@ def forward_block(self, key: LabelsEntry, block: TensorBlock) -> TensorBlock:
 
         out_values = module.forward(block.values)
         if self._out_properties is None:
-            properties = Labels.range("_", out_values.shape[-1])
+            # we do not use range because of metatensor/issues/410
+            properties = Labels(
+                "_",
+                _dispatch.arange_like(
```

Comment:
I think I like the solution used in metatensor-operations instead of a separate dispatch function: `_dispatch.int_array_like(range(shape)).reshape(-1, 1)`.

There is also the question of what happen when working with torch data inside metatensor-core; I'll have a look at this in a separate PR

Reply:
ah sorry did not see this comment till now, will do it

Reply:
done